File: BE/ClinicBooking/Models/Transaction.cs
using System.ComponentModel.DataAnnotations;

namespace ClinicBooking.Models
{
    public class Transaction : EntityBase
    {
        public int TransactionID { get; set; }
        public int MedicalHistoryId { get; set; }
        public MedicalHistory MedicalHistory { get; set; }

        [Required]
        public PaymentType PaymentType { get; set; }

        public bool Paid { get; set; }
        public DateTime? PaidDate { get; set; }
    }

    public enum PaymentType
    {
        Cash,
        Card,
        BankTransfer
    }
}

File: fe/types/transaction.ts
export type PaymentType = 'Cash' | 'Card' | 'BankTransfer';

export interface Transaction {
  TransactionID: number;
  medicalHistoryId: number;
  paymentType: PaymentType | number;
  paid: boolean;
  paidDate: string | null;
  active: boolean;
}

export interface TransactionResponse {
  data: Transaction[];
  pagination: {
    pageNumber: number;
    pageSize: number;
    totalItems: number;
  };
}

export interface TransactionFilter {
  fromDate?: string;
  toDate?: string;
  paymentType?: PaymentType | "";
  paid?: boolean | "";
  pageSize?: number;
  pageNumber?: number;
}